src/Lean/Meta/Tactic/Simp/Types.lean

```diff
@@ -45,9 +45,15 @@ structure State where
 
 abbrev SimpM := ReaderT Context $ StateRefT State MetaM
 
+structure SavedState where
+  meta : Meta.SavedState
+  usedTheorems : UsedSimps
+
 instance : MonadBacktrack SavedState SimpM where
-  saveState      := Meta.saveState
-  restoreState s := s.restore
+  saveState      := return ⟨← Meta.saveState, (← get).usedTheorems⟩
```

Comment:
Do you still need this modification?

Reply:
No, I don't. In fact, this instance is dead code which I'll separately propose deleting.